src/vs/platform/dialogs/electron-main/dialogMainService.ts

```diff
@@ -123,31 +124,45 @@ export class DialogMainService implements IDialogMainService {
 		return;
 	}
 
-	private getDialogQueue(window?: BrowserWindow): Queue<any> {
-		if (!window) {
-			return this.noWindowDialogQueue;
-		}
+	private getWindowDialogQueue<T>(window?: BrowserWindow): Queue<T> {
 
-		let windowDialogQueue = this.mapWindowToDialogQueue.get(window.id);
-		if (!windowDialogQueue) {
-			windowDialogQueue = new Queue<any>();
-			this.mapWindowToDialogQueue.set(window.id, windowDialogQueue);
-		}
+		// Queue message box requests per window so that one can show
+		// after the other.
+		if (window) {
+			let windowDialogQueue = this.windowDialogQueues.get(window.id);
+			if (!windowDialogQueue) {
+				windowDialogQueue = new Queue<T>();
+				this.windowDialogQueues.set(window.id, windowDialogQueue);
+			}
 
-		return windowDialogQueue;
+			return windowDialogQueue;
+		} else {
+			return this.noWindowDialogueQueue;
+		}
 	}
 
-	showMessageBox(options: MessageBoxOptions, window?: BrowserWindow): Promise<MessageBoxReturnValue> {
-		return this.getDialogQueue(window).queue(async () => {
-			if (window) {
-				return dialog.showMessageBox(window, options);
-			}
+	async showMessageBox(options: MessageBoxOptions, window?: BrowserWindow): Promise<MessageBoxReturnValue> {
 
-			return dialog.showMessageBox(options);
-		});
+		// prevent duplicates of the same dialog queueing at the same time
+		const fileDialogLock = await this.acquireFileDialogLock(options, window);
```

Comment:
I would not have expected the use of `fileDialogLock` inside the `showMessageBox` method. I would leave the newly introduced lock only for `showSaveDialog` and `showOpenDialog`

Reply:
Agreed, fixed in 7106e19